src/fetchServices.ts

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env node
+
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {google, servicemanagement_v1} from 'googleapis';
+const spreadsheetId = '14JYzl_8W3HyD0c1jYDXQJA2sVWBZiMoxT8Sp57xGWvk';
+
+/**
+ * Make a call to the Google Cloud Service Management API's services.list endpoint
+ * @return an array of hostnames ('foo.googleapis.com') for all available services
+ */
+export async function getAllServiceNames(): Promise<string[]> {
+  const auth = await google.auth.getClient({
+    scopes: ['https://www.googleapis.com/auth/cloud-platform'],
+  });
+  const client = google.servicemanagement({
+    version: 'v1',
+    auth: auth,
+  });
+  const serviceNames: string[] = [];
+  return await listServices(client, serviceNames);
+}
+
+export async function listServices(
+  client: servicemanagement_v1.Servicemanagement,
+  serviceNames: string[],
+  nextPageToken = ''
+): Promise<string[]> {
+  const res = await client.services.list(
+    {
+      pageSize: 100,
+      pageToken: nextPageToken,
+    },
+    {
+      retryConfig: {
+        retry: 10,
+      },
+    }
+  );
+
+  if (res.data.services) {
+    res.data.services.map(item => {
+      if (item.serviceName) {
+        serviceNames.push(item.serviceName);
+      }
+    });
+  }
+
+  if (res.data.nextPageToken) {
+    await listServices(client, serviceNames, res.data.nextPageToken);
+  }
+  return serviceNames;
+}
+
+/**
+ * Make a call to the Google Cloud Service Management API's services.getConfig endpoint
+ * @param serviceName - the hostname of a service ('foo.googleapis.com')
+ * @return the service config of the passed hostname
+ */
+export async function getServiceConfig(
+  serviceName: string
+): Promise<servicemanagement_v1.Schema$Service> {
+  const auth = await google.auth.getClient({
+    scopes: ['https://www.googleapis.com/auth/cloud-platform'],
+  });
+  const servicemanagement = google.servicemanagement({
+    version: 'v1',
+    auth: auth,
+  });
+  const res = await servicemanagement.services.getConfig(
+    {
+      serviceName: serviceName,
+    },
+    {
+      retryConfig: {
+        retry: 10,
+      },
+    }
+  );
+  return res.data;
+}
+
+/**
+ * Get service configs for all available services.
+ * @return the service configuration object
+ */
+export async function getAllServiceConfigs() {
+  const serviceNames = await getAllServiceNames();
+  const serviceConfigs = serviceNames.map(async (name: string) => {
```

Comment:
I don't think this will do what you want.  The map here is creating an array of promises, which you never wait to drain/complete.  You're also creating a double wrapping of async functions that isn't needed.  I suspect you want something like:

```js
const serviceConfigs = await Promise.all(
  serviceNames.map(name => getServiceConfig(name))
);
```